src/pages/preferences/preferencesTabs/NotificationsTab.tsx

```diff
@@ -23,13 +23,15 @@ const { omitTypename } = string;
 export const NotificationsTab: React.VFC = () => {
   const dispatchToast = useToastContext();
   const { userSettings, loading } = useUserSettings();
-  const { slackUsername, notifications } = userSettings ?? {};
+  const { slackUsername, slackMemberId, notifications } = userSettings ?? {};
   const [slackUsernameField, setSlackUsernameField] = useState(slackUsername);
+  const [slackMemberIdField, setSlackMemberIdField] = useState(slackMemberId);
   const [notificationStatus, setNotificationStatus] = useState(notifications);
   const { sendEvent } = usePreferencesAnalytics();
   // update state from query
   useEffect(() => {
     setSlackUsernameField(slackUsername);
+    setSlackMemberIdField(slackMemberId);
     setNotificationStatus(notifications);
   }, [slackUsername, notifications]);
```

Comment:
I think you need to add `slackMemberId` to the dependency array here 
```suggestion
  }, [slackUsername, slackMemberId, notifications]);
```